### PR TITLE
config(ticdc): Fix old value configuration check for maxwell protocol

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve? 
close https://github.com/pingcap/ticdc/issues/3676

### What is changed and how it works?

* add validate funcs
* complete protocol when create changefeed
* add tests

### Check List 

Tests 

 - Unit test
 - Manual test(https://github.com/pingcap/ticdc/pull/3747#issuecomment-986629511)
Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Side effects

None

Related changes

 - Need to cherry-pick to the release branch

### Release note 

```release-note
Fix old value configuration check for maxwell protocol.
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
Manual testing:
1. start tidb cluster
2. start cdc server 
3. add new changefeed config file
```toml
enable-old-value=false

[sink]

protocol = "maxwell"
```
4. run: `./cdc cli changefeed create --pd http://127.0.0.1:2379 --sink-uri kafka://127.0.0.1:9092/ticdc-test --config=changefeed.toml`
5. output:
```log
Global Flags:
      --ca string          CA certificate path for TLS connection
      --cert string        Certificate path for TLS connection
  -i, --interact           Run cdc cli with readline
      --key string         Private key path for TLS connection
      --log-level string   log level (etc: debug|info|warn|error) (default "warn")
      --pd string          PD address, use ',' to separate multiple PDs (default "http://127.0.0.1:2379")

[CDC:ErrKafkaInvalidConfig]kafka config invalid: maxwell protocol requires old value to be enabled

```

**Comment by hi-rustin** [comment]
/cc @3AceShowHand @amyangfei 

Could you please take a look? Thanks! 💚 💙 💜 💛 ❤️

File: pkg/config/sink.go [hunk 1]
```diff
@@ -30,3 +38,17 @@ type ColumnSelector struct {
...
+func (s *SinkConfig) validate(enableOldValue bool) error {
+	protocol := s.Protocol
+	if (protocol == ProtocolCanal.String() ||
+		protocol == ProtocolCanalJSON.String() ||
+		protocol == ProtocolMaxwell.String()) && !enableOldValue {
+		log.Error(fmt.Sprintf("Old value is not enabled when using `%s` protocol. "+
+			"Please update changefeed config", protocol))
+		return cerror.WrapError(cerror.ErrKafkaInvalidConfig,
+			errors.New(fmt.Sprintf("%s protocol requires old value to be enabled", protocol)))
+	}
```
Comment (amyangfei) [inline pkg/config/sink.go:51]: what about
```go
if !enableOldValue {
    switch protocol {
        case ProtocolCanal.String(), ProtocolCanalJSON.String(), ProtocolMaxwell.String():
            return error
    }
}

Comment (hi-rustin) [reply pkg/config/sink.go]: Addressed!

Comment (hi-rustin) [reply pkg/config/sink.go]: Thanks!

**Review by amyangfei — APPROVED** [review]

**Review by 3AceShowHand — APPROVED** [review]

**Pull request merged**